src/types/ical.js.d.ts

```diff
@@ -176,10 +176,19 @@ declare module 'ical.js' {
     public bysetpos?: number[];
   }
 
+  export class RecurIterator {
```

Comment:
For all of this file, please copy over the file how it is from the ios repo. This is already mostly (fully?) implemented there, so no need to implement it here differently.

Reply:
should I copy the entire file or just the relevant parts?

Reply:
Just looked at the code and it doesn't include the same typings that I added. Actually, I think this file was initially taken from the iOS repo because it is exactly the same (except for what I added). I had to add extra types that were needed for this code.

Reply:
Sure, you can add extra types, but first copy it over, because with your changes the file diverged, and you implemented some things differently.

Reply:
Sorry, but I am a little confused. As far as I can tell, the files were exactly the same before I added anything. And the things I have implemented here don't exist in the iOS file, so I'm not quite sure what I should be copying over?

The things I've implemented differently are purposefully different. I'm not sure if you are referring to the divergence of this `RecurIterator` vs the `RecurExpansion`, but these are intentionally different. I'm going off of the icaljs docs here. [`Event.iterator`](https://mozilla-comm.github.io/ical.js/api/ICAL.Event.html#iterator) and [`Recur.iterator`](https://mozilla-comm.github.io/ical.js/api/ICAL.Recur.html#iterator) return different objects. The former returns a `RecurExpansion` and the latter returns a `RecurIterator`. I am using the latter in this PR.

Reply:
My bad, sorry. I didn't notice this was on Recur instead of Event.